### PR TITLE
Move collapse out of plot and make it into a free function

### LINKED ISSUE #1217: Overplotting data from multiple objects or slices
As discussed in #802, users would like to plot data from multiple data arrays or dataset in the same figure. The problem is more general: Even if axes are shared, data may be in multiple objects. The solution proposed there is to use matplotlib:

```python
import matplotlib.pyplot as plt
fig, ax = plt.subplots(1, 1, figsize=(9, 6))
plot(d1["Sample1"], mpl_axes=ax)
plot(d2["Sample2"], mpl_axes=ax, color=1)
ax.set_ylim([0, 10])
fig
```

However, in practice it required a fair bit more work, non obvious to a user:

```python
import matplotlib.pyplot as plt
fig, ax = plt.subplots(1, 1, figsize=(7, 5))

plot(db, logy=True, collapse='Q', mpl_axes=ax)

plot(db_full, color=9, mpl_axes=ax)
ax.lines[-1].set_linestyle('dotted') # because `linestyle` option is ignored for histogram
ax.lines[-1].set_label('full range') # because there is no `label` option to `plot`

plot(db_fit, mpl_axes=ax)
ax.lines[-1].set_linestyle('dashed') # see above
ax.lines[-1].set_label('fit') # see above

ax.set_ylim([0.1, 100])
ax.legend() # because setting labels by hand apparently does not update legend
fig.show()
```

See inline comments for required changes. A more convenient wrapper for this "overplotting" application is likely a requirement. Fixing some of the inconveniences (non-working linestyle and label options) may get us part of the way, but "native" overplotting support should still be considered.

A second, related but different, application is plotting selected indices from an object. Scientists would like to plot, e.g., slices `0,3,9,17` from an object. This is not supported by our slicing syntax. We might use a loop to plot this to the same axes:
```python
import matplotlib.pyplot as plt
fig, ax = plt.subplots(1, 1, figsize=(7, 5))
for i, straw in enumerate([1,3,4]):
    plot(straw_counts['straw',straw], color=i, marker=i, mpl_axes=ax)
```
This is actually relatively convenient already, except for the need to `enumerate` to manually set the color and marker (otherwise everything has identical colors and markers). However, this is still missing the legend, so we have to do this by hand again, which brings us to the more complex:
```python
import matplotlib.pyplot as plt
fig, ax = plt.subplots(1, 1, figsize=(7, 5))
i = 0
for tube, straw in (1,3),(3,5),(4,0),(11,3):
    plot(pixel_counts['tube', tube]['straw', straw],
         color=i, marker='.',
         mpl_axes=ax)
    ax.lines[-1].set_label(f'tube-{tube}-straw{straw}')
    i += 1
ax.legend() # last label does not show without this
```

### PR DESCRIPTION
**Pull request by nvaytet**
This removes the
```Py
plot(d, collapse='dim_to_keep')
```
syntax, in favor of a more generic
```Py
plot(sc.collapse(d, keep='dim_to_keep'))
```

In addition, a very similar function `sc.slices` was added:
- `collapse(d, keep='x')` slices all dimensions away to keep only `'x'`, thus always returning 1D slices.
- `slices(d, dim='x')` slices along dimension `'x'`, returning slices with `ndim-1` dimensions contaiing all dimensions other than `'x'`.

The slices are returned in a python `dict`.

Some examples on how to plot using a dict of DataArrays was also added to the `plotting-1d-data` notebook in the docs.

Fixes #1217

### CODE REVIEW

File: docs/user-guide/slicing.ipynb [hunk 1]
```diff
@@ -198,6 +201,76 @@
     "d['x', 1:2]['a'] == d['a']['x', 1:2]\n",
     "d['x', 1:2]['a'].coords['x'] == d.coords['x']['x', 1:2]"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Slicing tools\n",
```
Comment (SimonHeybrock) [inline docs/user-guide/slicing.ipynb:209]: I am not sure this is the best place for this documentation. As it was, the slicing page introduced the fundamental concepts essential for the understanding of how to use scipp, whereas this section is about specialized routines that are non-essential. Should this rather go into the documentation of these methods?

Comment (nvaytet) [reply docs/user-guide/slicing.ipynb:209]: Fine by me. So we leave the example in the 1D plotting, and then just rely on the sphinx documentation of the API?

Comment (SimonHeybrock) [reply docs/user-guide/slicing.ipynb:209]: I suppose you can move the code examples to the sphinx docstrings?

File: python/tests/utils/test_to_string.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Neil Vaytet
+
+import numpy as np
+import scipp as sc
+import scipp.utils as su
```
Comment (SimonHeybrock) [inline python/tests/utils/test_to_string.py:8]: I am not sure we should introduce a new submodule into scipp without more curation/triaging of what goes in. Is there a way to keep this more private and not part of scipp?

Comment (nvaytet) [reply python/tests/utils/test_to_string.py:8]: yes i can do that

Comment (nvaytet) [reply python/tests/utils/test_to_string.py:8]: it seems if i don't want these low-level functions to be part of a submodule, I may have to introduce some `sys.path.append(...)` stuff in here.
Is that ok or should I look for another solution?
e.g.
```Py
import sys
sys.path.append('../src/scipp/utils')
```

Comment (SimonHeybrock) [reply python/tests/utils/test_to_string.py:8]: Maybe just make it `scipp._util`, following the "convention" for marking things as private?

Comment (SimonHeybrock) [reply python/tests/utils/test_to_string.py:8]: See also https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces.

**Pull request merged**